diff --git a/sdk_core/CMakeLists.txt b/sdk_core/CMakeLists.txt
index 15e960d..93f31d1 100644
--- a/sdk_core/CMakeLists.txt
+++ b/sdk_core/CMakeLists.txt
@@ -29,13 +29,13 @@ else(WIN32)
 endif (WIN32)

 target_compile_options(${SDK_LIBRARY_STATIC}
-        PRIVATE $<$<CXX_COMPILER_ID:GNU>:-Wall>#-Wno-c++11-long-long>
+        PRIVATE $<$<CXX_COMPILER_ID:GNU>:-Wall -Wno-pedantic -Wno-pragmas>#-Wno-c++11-long-long>
         PRIVATE $<$<CXX_COMPILER_ID:AppleClang>:-Wno-unknown-pragmas -Wall -Werror -Wno-c++11-long-long>
         PRIVATE $<$<CXX_COMPILER_ID:Clang>:-Wno-unknown-pragmas -Wall -Werror -Wno-c++11-long-long>
         )

 target_compile_options(${SDK_LIBRARY_SHARED}
-        PRIVATE $<$<CXX_COMPILER_ID:GNU>:-Wall>#-Wno-c++11-long-long>
+        PRIVATE $<$<CXX_COMPILER_ID:GNU>:-Wall -Wno-pedantic -Wno-pragmas>#-Wno-c++11-long-long>
         PRIVATE $<$<CXX_COMPILER_ID:AppleClang>:-Wno-unknown-pragmas -Wall -Werror -Wno-c++11-long-long>
         PRIVATE $<$<CXX_COMPILER_ID:Clang>:-Wno-unknown-pragmas -Wall -Werror -Wno-c++11-long-long>
         )
diff --git a/sdk_core/base/multiple_io/multiple_io_epoll.cpp b/sdk_core/base/multiple_io/multiple_io_epoll.cpp
index d5b709c..1e44476 100644
--- a/sdk_core/base/multiple_io/multiple_io_epoll.cpp
+++ b/sdk_core/base/multiple_io/multiple_io_epoll.cpp
@@ -59,7 +59,7 @@ bool MultipleIOEpoll::PollSetAdd(PollFd poll_fd) {
   if (max_poll_size_ <= (int)descriptors_.size()) {
     return false;
   }
-  struct epoll_event ee = {0};
+  struct epoll_event ee = {};
   ee.events = GetEvent(poll_fd.event);
   ee.data.fd = poll_fd.fd;
   if (epoll_ctl(epoll_fd_, EPOLL_CTL_ADD, poll_fd.fd, &ee) == -1) {
@@ -73,7 +73,7 @@ bool MultipleIOEpoll::PollSetAdd(PollFd poll_fd) {

 bool MultipleIOEpoll::PollSetRemove(PollFd poll_fd) {
   int fd = poll_fd.fd;
-  struct epoll_event ee = {0};
+  struct epoll_event ee = {};
   epoll_ctl(epoll_fd_, EPOLL_CTL_DEL, fd, &ee);
   if (descriptors_.find(fd) != descriptors_.end()) {
       descriptors_.erase(fd);
diff --git a/sdk_core/upgrade/firmware.h b/sdk_core/upgrade/firmware.h
index c39f22c..409b950 100644
--- a/sdk_core/upgrade/firmware.h
+++ b/sdk_core/upgrade/firmware.h
@@ -119,7 +119,7 @@ class Firmware {
   bool Open(const char *firmware_path);
   void Close();

-  const uint32_t FirmwarePackageVersion() const { return header_.file_version; }
+  uint32_t FirmwarePackageVersion() const { return header_.file_version; }


   LivoxEncryptFirmwareHeader header_;
